lib/project_copier.rb

```diff
@@ -0,0 +1,30 @@
+class ProjectCopier
+  EXCLUDE_ATTRIBUTES = [ :classifications_count, :launched_row_order, :beta_row_order ].freeze
+  INCLUDE_ATTRIBUTES = [  :project_contents,
+                          :tutorials,
+                          :field_guides,
+                          :pages,
+                          :tags,
+                          :tagged_resources,
+                          :avatar,
+                          :background,
+                          { active_workflows: [ :tutorials, :attached_images, :workflow_contents ] }
+                        ].freeze
+
+  def self.copy(project_id, user_id)
+    project = Project.find(project_id)
+    user = User.find(user_id)
+
+    copied_project = project.deep_clone include: INCLUDE_ATTRIBUTES, except: EXCLUDE_ATTRIBUTES
+    copied_project.owner = user
```

Comment:
did you look at the block syntax for updating the clone? https://github.com/moiristo/deep_cloneable#optional-block

Reply:
I'm not doing anything with the copy that needs any pre-exclude original attributes or anything else I can think of that requires that block. It's all happening pre-`save!`, so it's not hitting the db yet.